shell/utils/auth.js

```diff
@@ -315,6 +315,10 @@ export function isLoggedIn(store, me) {
 export function notLoggedIn(store, redirect, route) {
   store.commit('auth/hasAuth', true);
 
+  if (!route.name.includes('auth')) {
+    store.commit('prefs/setAuthRedirect', route);
+  }

```

Comment:
This area of the code is responsible for redirecting the user to the login page if they aren't logged in. The perfect place to get the url that was used before redirecting to login.